omniscidb/DataMgr/FixedLengthEncoder.h

```diff
@@ -91,9 +91,10 @@ class FixedLengthEncoder : public Encoder {
           for (size_t i = range.begin(); i < range.end(); i++) {
             if (data[i] != inline_null_value<V>()) {
               if (!fixlen_array || data[i] != inline_null_array_value<V>()) {
-                decimal_overflow_validator_.validate(data[i]);
-                min = std::min(min, data[i]);
-                max = std::max(max, data[i]);
+                if (decimal_overflow_validator_.validate(data[i])) {
```

Comment:
Do we just ignore out-of-range decimal values now or do we assume it's checked on the arrow decimal conversion? Shouldn't this validation always succeed?

Reply:
Well, I am skeptical we ever hit this branch but can't guarantee it. So, we ignore out of range values. I suppose this could pose a problem later, and maybe we should just remove this code altogether and make sure we validate the decimal range on import. Alternatively we could try aborting with an error message (or perhaps an exception) and not specify whether the decimal is above or below the range. It is silly to pay the cost of a function call for every single value in every single integer column, even if that function just returns.

Reply:
On one side, making a range check combined with stats computation is very reasonable and reduces the number of scans. On the other side, our only external data source is Arrow, which uses another decimal format and always requires conversion. So, we can probably make the converter responsible for the check and remove the decimal validator from encoders.